### PR TITLE
Enable keyboard shortcut to mute from any tab (no perms)

### PR DESCRIPTION
**Pull request by nedarb**
Allow a global (throughout all browser windows) mute/unmute toggle keyboard shortcut

Note: this does not add any new extension permissions, unlike PR #7 did

### CODE REVIEW

File: src/background.js [hunk 1]
```diff
@@ -12,3 +12,21 @@ chrome.runtime.onInstalled.addListener(function () {
...
+  if (command === "toggle-meeting-mute" && ports.size > 0) {
+    for (const port of ports) {
+      port.postMessage({ toggle: "mute" });
+    }
+  }
+});
+
+// add a handler for when a Google Meet content script loads
+chrome.runtime.onConnect.addListener(function (port) {
+  console.assert(port.name == "meet");
```
Comment (tpott) [inline src/background.js]: nit: add a message to this assertion, something like `"Unexpected content script connection %s", port.name)`

File: src/background.js [hunk 2]
```diff
@@ -12,3 +12,21 @@ chrome.runtime.onInstalled.addListener(function () {
     ]);
   });
 });
+
+// keep a list of connected content scripts so that we can tell them to toggle mute
+const ports = new Set();
+// listen to global keyboard handler events
+chrome.commands.onCommand.addListener(function (command) {
+  if (command === "toggle-meeting-mute" && ports.size > 0) {
```
Comment (tpott) [inline src/background.js]: nit: I'd invert this and return early. I.e. `if (command !== 'toggle' || ports.size === 0) { return; }` (but using newlines)

**Comment by tpott** [comment]
https://media.giphy.com/media/48FhEMYGWji8/giphy.gif

**Comment by nedarb** [comment]
Thanks @tpott - I tweaked those things you mentioned

**Comment by tpott** [comment]
Thanks! Can you add @stopdropandrew as a reviewer? I'm not associated with BashVideo, so I can't accept PRs

**Comment by nedarb** [comment]
@stopdropandrew can I get code review of this?

**Comment by at-avacron** [comment]
I'm not a programmer, but background functionality is a big deal for me. I have my hotkey set to **shift**. There's a Chrome-only web app that I need to run at the same time as Meet, whose mute/unmute hotkey can only be **shift**, and whose tab has to be in focus to receive that command. If GMPTT could address a background Meet tab, I could use one key to simultaneously unmute GMPTT and mute the other app.

File: src/manifest.json [hunk 3]
```diff
@@ -14,6 +14,15 @@
     }
   ],
   "background": { "scripts": ["background.bundle.js"], "persistent": false },
+  "commands": {
+    "toggle-meeting-mute": {
+      "suggested_key": {
+        "default": "Ctrl+Space",
+        "mac": "Ctrl+Space"
```
Comment (stopdropandrew) [inline src/manifest.json]: Can you change this to `"MacCtrl+Space"` ? On Mac cmd-space opens spotlight search.

File: package.json [hunk 4]
```diff
@@ -19,6 +20,7 @@
   "devDependencies": {
     "@babel/core": "^7.9.0",
     "@babel/preset-env": "^7.9.5",
+    "@types/chrome": "^0.0.114",
```
Comment (stopdropandrew) [inline package.json]: Is this needed? No typescript here

File: src/ptt.js [hunk 5]
```diff
@@ -60,3 +60,12 @@ getSavedValues(({ hotkey, muteOnJoin }) => {
 addChangeListener(({ hotkey }) => {
   hookUpListeners(hotkey);
 });
+
+var port = chrome.runtime.connect({ name: "meet" });
+port.onMessage.addListener(function (msg) {
+  if (msg?.toggle === "mute") {
+    const offButton = document.querySelector(micButtonSelector(MIC_OFF));
+    const onButton = document.querySelector(micButtonSelector(MIC_ON));
```
Comment (stopdropandrew) [inline src/ptt.js:77]: Can you change this to work with the new language support?

**Review by stopdropandrew — CHANGES_REQUESTED** [review]
Sorry about the delay on this, been busy. Mostly looks great, I had to change the mac hotkey to make it work. I really wish we could track keydown/up and make this PTT but a toggle is better than nothing. Great work! I won't let it sit long this time 😅

**Comment by nedarb** [comment]
@stopdropandrew I think I addressed your comments in the latest commits, if you have time to take a look

**Comment by karlhorky** [comment]
Ah this would be cool! @stopdropandrew do you think this could get merged?

**Pull request merged**